lib/marked.js

```diff
@@ -27,8 +27,8 @@ var block = {
     + '|<![A-Z][\\s\\S]*?>\\n*' // (4)
     + '|<!\\[CDATA\\[[\\s\\S]*?\\]\\]>\\n*' // (5)
     + '|</?(tag)(?: +|\\n|/?>)[\\s\\S]*?(?:\\n{2,}|$)' // (6)
-    + '|<(?!script|pre|style)([a-z][\\w-]*)(?:attribute)*? */?>(?=\\h*\\n)[\\s\\S]*?(?:\\n{2,}|$)' // (7) open tag
-    + '|</(?!script|pre|style)[a-z][\\w-]*\\s*>(?=\\h*\\n)[\\s\\S]*?(?:\\n{2,}|$)' // (7) closing tag
```

Comment:
@Feder1co5oave I'm not sure what the `\\h*` is supposed to do. As far as I know `\H` isn't a special character.

Reply:
`\h` has meaning in some languages.

| Language | Meaning of `\h` |
| ------------- | --------------- |
| Ruby | Hex chars, e.g. `[0-9a-fA-F]` |
| Java, PHP, Perl | Horizontal whitespace, cf. `\v` |

Reply:
...so I imagine this regex originated in a language (or in someone's head) where `\h` had meaning.

Reply:
...I've written a research paper on this topic if @UziTech @styfle @Feder1co5oave would be interested.

Reply:
- Is there a test case that includes horizontal whitespace where the \h was?
- It's possible the original regex should have used something like [ \t] instead?

Reply:
It looks like it should be `[ \\t]*` but I would like to get @Feder1co5oave point of viewq since he is the one that introduced it.

Reply:
It was intended to be `[ \t]*` I guess, I just don't remember why I used `\h` instead, and why it required a newline after the tag :(

Reply:
I added `[ \t]*` in place of `\h` but with some manual testing it doesn't seem to change anything. All of the tests I can think of still pass with or without `[ \t]*`

Reply:
That could be due to the fact that we strip white space before we check the output.

Reply:
We still strip whitespace when testing? Wasn't html-differ supposed to fix this?

Reply:
We were only using html-differ for the spec tests. The files were still using the manual method.

I fixed it in #1449